apps/web/src/pages/SearchPage.tsx

```
@@ -124,25 +219,44 @@ export function SearchPage() {
+    if (last.query === filters.query && last.filter === filters.filter && rawResults !== null) return;
+
     if (debounceRef.current) clearTimeout(debounceRef.current);
     debounceRef.current = setTimeout(() => {
-      void doSearch(filter, query);
+      void doSearch(filters.filter, filters.query);
     }, 300);
     return () => {
       if (debounceRef.current) clearTimeout(debounceRef.current);
     };
-  }, [query, filter, doSearch]);
+  }, [filters.query, filters.filter, doSearch]); // eslint-disable-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)